src/compiler/types.ts

```diff
@@ -6718,6 +6718,7 @@ namespace ts {
         readonly importModuleSpecifierEnding?: "auto" | "minimal" | "index" | "js";
         readonly allowTextChangesInNewFiles?: boolean;
         readonly providePrefixAndSuffixTextForRename?: boolean;
+        readonly stringLiteralReplacementMode?: boolean;

```

Comment:
@amcasey @uniqueiniquity  @mjbvz  do we really need this preference for compatibility given that the replacementSpan is existing property.

Reply:
Nope - if there's a replacementSpan, we'll handle it.

Reply:
Not my opinion, but I don't want to ruin anything

Reply:
we need to wait on @mjbvz to confirm if this is needed for vscode.. Otherwise this flag is not needed and shouldnt be added.

Reply:
Does the flag is added for the user who wants to keep the classic behavior, IMO...
I‘m not 100% sure everyone like the replacement.🤔